rust/windows-client/src-tauri/src/client.rs

```diff
@@ -102,26 +102,33 @@ pub(crate) fn run() -> Result<()> {
 ///
 /// Automatically logs or shows error dialogs for important user-actionable errors
 fn run_gui(cli: Cli) -> Result<()> {
-    let result = gui::run(cli);
+    let result = gui::run(&cli);
 
     // Make sure errors get logged, at least to stderr
-    if let Err(error) = &result {
+    if let Err(error) = result {
         tracing::error!(?error, "gui::run error");
-        let error_msg = match &error {
-            gui::Error::WebViewNotInstalled => "Firezone cannot start because WebView2 is not installed. Follow the instructions at <https://www.firezone.dev/kb/user-guides/windows-client>.".to_string(),
-            error => format!("{}", error),
-        };
-
-        native_dialog::MessageDialog::new()
-            .set_title("Firezone Error")
-            .set_text(&error_msg)
-            .set_type(native_dialog::MessageType::Error)
-            .show_alert()?;
+
+        if let Some(Cmd::SmokeTest) = &cli.command {
```

Comment:
We wouldn't need this branch if we follow the structure I laid out in the previous comment, however, if we keep this we can just:
```rs
if let Some(Cmd::SmokeTest) = &cli.command { return Ok(()); }
```